### PR TITLE
PLG-34: excluding orders without email

### PR DESCRIPTION
**Pull request by nedick007**

### CODE REVIEW

File: app/code/community/Metrilo/Analytics/Helper/Data.php [hunk 1]
```diff
@@ -265,7 +265,7 @@ private function _buildOrdersForSubmition($orders) {
         $ordersForSubmition = array();
 
         foreach ($orders as $order) {
-            if ($order->getId() && $order->getStatus() != null) {
+            if ($order->getId() && $order->getStatus() != null && $order->getCustomerEmail()) {
```
Comment (evgeni-spasov-metrilo) [inline app/code/community/Metrilo/Analytics/Helper/Data.php]: What if the email is whitespace?

Comment (nedick007) [reply app/code/community/Metrilo/Analytics/Helper/Data.php]: Usually checkout fields have js validators assigned.

Comment (evgeni-spasov-metrilo) [reply app/code/community/Metrilo/Analytics/Helper/Data.php]: Be prepared to get some kicks from the new tracking with field validations 🗡

Comment (adamof) [reply app/code/community/Metrilo/Analytics/Helper/Data.php]: is `'' == true` in PHP?

Comment (nedick007) [reply app/code/community/Metrilo/Analytics/Helper/Data.php]: Using Loose comparisons with == in PHP results as FALSE for empty string like '' and True for empty space like ' '.

Comment (adamof) [reply app/code/community/Metrilo/Analytics/Helper/Data.php]: is there anything that can strip whitespace and then check presence?

Comment (nedick007) [reply app/code/community/Metrilo/Analytics/Helper/Data.php]: trim() method is removing white spaces. As evgeni mentioned there is going to be troubles with validation, but that should come before the data is inserted in the database. Adding this check 'trim($order->getCustomerEmail())' will ensure there are no errors when excluding empty mail orders. I assumed it's not necessary but will add it in the following commit.

**Comment by adamof** [comment]
@nedick007 is this ready for review?

**Comment by nedick007** [comment]
Yes, code is ready for review.

**Review by adamof — APPROVED** [review]

**Pull request merged**